File: solutions/javascript/sum-of-multiples/1/sum-of-multiples.js
/**
 * Calculates the sum of all multiples of the given numbers below a specified limit.
 *
 * @param {number[]} multiples - An array of numbers to find multiples of.
 * @param {number} limit - The upper limit (exclusive) up to which to calculate the sum.
 * @returns {number} - The sum of all multiples of the given numbers below the limit.
 */
export function sum(multiples, limit) {
  let result = 0;
  for (let index = 1; index < limit; index++) {
    if (multiples.some((multiple) => index % multiple === 0)) {
      result += index;
    }
  }
  return result;
};

File: solutions/lua/sum-of-multiples/1/sum-of-multiples.lua
local function sum_of_multiples(numbers)
  return {
    to = function(number)
        local sum = 0
        for index1=1,number-1 do
          for _, value in ipairs(numbers) do
            if index1 % value == 0 then
              sum = sum + index1
              break
            end
          end
        end
        return sum
      end
  }
end

return sum_of_multiples